identity-iota/src/credential/errors.rs

```diff
@@ -0,0 +1,138 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use std::collections::BTreeMap;
+use std::fmt::Display;
+
+use identity_core::common::OneOrMany;
+
+use crate::did::IotaDIDUrl;
+
+#[derive(Debug, thiserror::Error)]
+#[non_exhaustive]
+/// An error associated with validating credentials and presentations.
+pub enum ValidationError {
+  /// Indicates that the expiration date of the credential is not considered valid.
+  #[error("the expiration date does not satisfy the validation criterea")]
+  ExpirationDate,
+  /// Indicates that the issuance date of the credential is not considered valid.
+  #[error("the issuance date does not satisfy the validation criterea")]
```

Comment:
"The credential is yet to be active as it hasn't passed the issuance date" for example could work here as well.

Reply:
Updated the descriptions now.